randomizer.py

```diff
@@ -381,7 +381,7 @@ def randomize(self):
         tweaks.remove_title_and_ending_videos(self)
       if self.options.get("remove_music"):
         patcher.apply_patch(self, "remove_music")
-      if self.map_select:
+      if self.map_select and not self.options.get("do_not_generate_spoiler_log"):
         patcher.apply_patch(self, "map_select")
```

Comment:
It might be less confusing if this throws an exception rather than just ignoring the map select patch?